numpy/f2py/crackfortran.py

```diff
@@ -2353,12 +2349,12 @@ def get_sorted_names(vars):
     """
     depend_dict = _calc_depend_dict(vars)
     names = []
-    for name in list(depend_dict.keys()):
+    for name in depend_dict:
```

Comment:
This is not safe - this line is deliberately copying the keys, as it modifies them during iteration.

Reply:
Absolutely -- reverted. Thanks.